### PR TITLE
fix: Migrate to eslint 9

### PR DESCRIPTION
**Pull request by Koenkk**
TODO:
- [x] Fix TODOS in `eslint.config.mjs`
- [x] Fix unnecessary `// eslint-disable` comments

CC: @Nerivec

### CODE REVIEW

File: scripts/modernExtendRefactor.ts [hunk 1]
```diff
@@ -60,10 +60,10 @@ project.getSourceFiles().forEach((sourceFile) => {
                     const evalOpts = Object.entries(eval(`(${opts})`));
                     for (const [key, value] of evalOpts) {
                         if (key === 'colorTempRange') {
-                            // @ts-expect-error
+                            // @ts-expect-error ignore
                             newOpts.colorTemp = {...newOpts.colorTemp, range: value};
```
Comment (Nerivec) [inline scripts/modernExtendRefactor.ts:64]: `newOpts.colorTemp as object` to get rid of the `@ts-expect-error`

File: src/converters/fromZigbee.ts [hunk 2]
```diff
@@ -2357,7 +2357,7 @@ const converters1 = {
         convert: (model, msg, publish, options, meta) => {
             const dp = msg.data[10];
             const defaults = {motor_direction: 'FORWARD', motor_speed: 40};
-            // @ts-expect-error
+            // @ts-expect-error ignore
             if ((msg.data[0] === 0x7a) & (msg.data[1] === 0xd1)) {
```
[the remark this answers is not in the document]
Comment (Koenkk) [reply src/converters/fromZigbee.ts:2364]: Looks to be wrong indeed, fixed!

File: scripts/modernExtendRefactor.ts (hunk 1 above)
Comment (Koenkk) [reply scripts/modernExtendRefactor.ts:64]: Done!

**Comment by Koenkk** [comment]
Z2M tests still pass with these changes 😄

**Comment by Alick116** [comment]
When will the device be supported? [MS-21-Z]

**Pull request merged**